### DISCUSSION TITLE
HIP-12: Zero Knowledge Proofs

### DISCUSSION THREAD

**Discussion by kenthejr** [hiero-ledger/hiero-improvement-proposals#52] [Edits]
This discussion is designated as the `discussions-to` target of [hip-12](https://github.com/hashgraph/hedera-improvement-proposal/blob/master/HIP/hip-12.md). Use this thread to further develop the HIP and move it through to its final lifecycle state.

**Comment by jimthereaper** [comment]
This would be awesome.

**Comment by ssundara2** [comment]
I would like to augment this proposal for NIZKP on Hedera Hashgraph to include scalable payment channels and post quantum encryption, DAML smart contracts.

Among the NIZKP's I would recommend zkSTARKs using the Starkware libraries from Dr Eli Ben Sasson
https://github.com/starkware-libs/ethSTARK
The scalable payment channels over lightning proposed by Starkware above also seem useful.

Beyond NIZKPs I would like to add post-quantum encryption to Hedera event messages using lattice based cryptography.
It may be nice to consider https://eprint.iacr.org/2016/870 for bootstrapping free FHE over lattices using the SWIFFT and PALISADE libraries at https://gitlab.com/palisade/palisade-release

For smart contracts DAML support would be wonderful https://github.com/digital-asset/daml

Thanks
Srikanth

**Comment by topshef** [comment]
This sounds great. I read up on ZKP once and thought that's amazing, but then got dizzy and fell back into ignorance :-/
A non-tech journalist friend even mentioned ZKP so I guess it's bubbling up :-)   

More education on this and how it could fit within business workflows might help push things. 

Any more material here would be amazing, wiki seems pretty good though https://en.wikipedia.org/wiki/Zero-knowledge_proof

**Comment by jodafm** [comment]
This HIP is amazing, any progress on this?

**Comment by ssundara2** [comment]
Hi Jose

We do not know the status of the proposal at the moment though Navnit 
and Jay on copy could help deploy resources

HIP-12: Zero Knowledge Proofs · Discussion #52 · 
hashgraph/hedera-improvement-proposal · GitHub [1]

Thanks

Srikanth

On 2021-08-24 11:08, José Sánchez wrote:

> This HIP is amazing, any progress on this?
> 
> --
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub [1], or unsubscribe 
> [2].
> Triage notifications on the go with GitHub Mobile for iOS [3] or 
> Android [4].


Links:
------
[1] 
https://github.com/hashgraph/hedera-improvement-proposal/discussions/52#discussioncomment-1228930
[2] 
https://github.com/notifications/unsubscribe-auth/ACU5RETBTZ5XJMLTYIZV47TT6PNYLANCNFSM4ZBE5CZA
[3] 
https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&amp;mt=8&amp;pt=524675
[4] 
https://play.google.com/store/apps/details?id=com.github.android&amp;utm_campaign=notification-email

**Comment by sergmetelin** [comment]
@rahul-kothari hi there! 
I wanted to check in on the status of this HIP - was there any further development of it? The HIP needs a bit more definition specifically related to Hedera's architecture in the `Specification` section.

**Reply by MULTICOMPLEX** [reply]
https://github.com/gakonst/awesome-starknet ?
